launcher/launcher.cpp

```diff
@@ -307,9 +307,26 @@ path GetHomedirPath() {
 	return buffer;
 }
 
+path GetDepotPath() {
+	string buffer;
+	buffer.resize(MAX_PATH);
+	DWORD pathLen = MAX_PATH;
+	
+	pathLen = GetEnvironmentVariableA("JULIA_DEPOT_PATH", &buffer[0], pathLen);
```

Comment:
I'm wondering whether we could use `std::getenv` here? If that works on Windows, then we could make this cross platform right away? It might look like this:
```cpp
const char *tmp = std::getenv("JULIA_DEPOT_PATH");

if(tmp) {
  string env_var {tmp};
  // Do the parsing etc here.
}
else {
  return GetHomedirPath() / ".julia";
}
```

Reply:
Yes, I agree this will immediately be more platform independent. What to do then about the `;`/`:` split? Maybe we can get away with a
```
#ifdef _WIN32
    #define ENVPATHSEP ';'
#else
    #define ENVPATHSEP ':'
#endif
```
But it would be nicer if we don't need a directive.

Reply:
```
const char *tmp = std::getenv("JULIA_DEPOT_PATH");

if(tmp) {
  string env_var {tmp};

  auto cutoff = env_var.find(ENVPATHSEP);
  if(cutoff != std::string::npos){
    env_var.resize(cutoff);
  }
  
  if(env_var.size() == 0){
      return env_var;
  }
}
  
return fallback;
```

Reply:
Maybe something like this? I can't really test this at the moment, so I'll just leave it here as a comment. (Maybe I'll get back to this sometime next week, but wouldn't be able to itch my FOSS addiction for the rest of this week at least)

Reply:
Okay, I added some cross-platform code that work on Windows, but haven't been tested on Linux yet.